pyanaconda/modules/subscription/satellite.py

```diff
@@ -0,0 +1,161 @@
+PROVISIONING_SCRIPT_DOWNLOAD_PATH = "/tmp/katello-rhsm-consumer"
+
+
+def download_satellite_provisioning_script(satellite_url, proxy_url=None):
+    """Download provisioning script from a Red Hat Satellite instance.
+
+    Download the provisioning script from a Satellite instance and store it
+    to a local file.
+
+    Satellite instances usually have self signed certificates and also some tweaks
+    are usually required in rhsm.conf to connect to a customer run Satellite instance
+    instead of to Hosted Candlepin for subscription purposes.
+
+    Each Satellite instance thus hosts a provisioning script available over plain
+    HTTP that client machines ca fetch and execute. This script has minimal dependencies
+    and provisions the machine to be able to talk to the one given Satellite instance
+    by installing it's self signed certificates and adjusting rhsm.conf.
+
+    NOTE: As the script is downloaded over plain HTTP is is adviced to ever only
+          provision machines from a Satellite instance on a trusted network, to
+          avoid the possibility of the provisioning script being tempered with
+          during transit.
+
+
+    :param str satellite_url: Satellite instance URL
+    :param proxy_url: proxy URL to use when fetchign the script
+    :type proxy_url: str or None if not set
+    :returns: True on success, False otherwise
+    """
+
+    # construct the URL pointing to the provisioning script
+    # FIXME: do this more clenly ? include the protocol in the URL ?
+    script_url = "https://" + satellite_url + PROVISIONING_SCRIPT_SUB_PATH
+
+    log.debug("subscription: fetching Satellite provisioning script from: %s", script_url)
+
+    headers = {"user-agent": USER_AGENT}
+    proxies = {}
+    provisioning_script = ""
+
+    if proxy_url is not None:
+        try:
+            proxy = ProxyString(proxy_url)
+            proxies = {"http": proxy.url,
+                       "https": proxy.url}
+        except ProxyStringError as e:
+            log.info("subscription: failed to parse proxy when fetching Satellite"
+                     " provisioning script %s: %s",
+                     proxy_url, e)
+
+    session = util.requests_session()
+
+    try:
+        # NOTE: we explicitely don't verify SSL certificates while
+        #       downloading the provisioning script as the Satellite
+        #       instance will have it's own self signed certs that
+        #       will only be trusted once the provisioning script runs
+        result = session.get(script_url, headers=headers,
+                             proxies=proxies, verify=False,
+                             timeout=constants.NETWORK_CONNECTION_TIMEOUT)
+        if result.ok:
+            provisioning_script = result.text
+            result.close()
+        else:
+            log.debug("subscription: server returned %i code when downloading"
+                      " Satellite provisioning script", result.status_code)
+            result.close()
```

Comment:
Please, call `return False` after this statement and simplify the code below.

Reply:
Yep, let's do that. :)